django/db/migrations/state.py

```diff
@@ -213,23 +212,14 @@ def __init__(self, real_apps, models, ignore_swappable=False):
 
         self.render_multiple(list(models.values()) + self.real_models)
 
-        # If there are some lookups left, see if we can first resolve them
-        # ourselves - sometimes fields are added after class_prepared is sent
-        for lookup_model, operations in self._pending_lookups.items():
-            try:
-                model = self.get_model(lookup_model[0], lookup_model[1])
-            except LookupError:
-                app_label = "%s.%s" % (lookup_model[0], lookup_model[1])
-                if app_label == settings.AUTH_USER_MODEL and ignore_swappable:
-                    continue
-                # Raise an error with a best-effort helpful message
-                # (only for the first issue). Error message should look like:
-                # "ValueError: Lookup failed for model referenced by
-                # field migrations.Book.author: migrations.Author"
-                msg = "Lookup failed for model referenced by field {field}: {model[0]}.{model[1]}"
-                raise ValueError(msg.format(field=operations[0][1], model=lookup_model))
-            else:
-                do_pending_lookups(model)
+        # There shouldn't be any operations pending at this point.
+        pending_models = set(self._pending_operations)
+        if ignore_swappable:
+            pending_models -= {make_model_tuple(settings.AUTH_USER_MODEL)}
+        if pending_models:
+            msg = "Unhandled pending operations for models: %s"
```

Comment:
Yeah, it is more informative – I initially catered for this by having ``Apps.lazy_model_operation`` take kwargs, but that bit of introspection was removed after some discussion in the previous PR: https://github.com/django/django/pull/3984#discussion-diff-24385523

It might be possible to get the field by introspecting the passed-in function object to see its partially-applied kwargs, but I think it would be a bit of fairly arcane code for the sake of this error message, and there are some usages which aren't related to a field at all. Would you be OK with this going in as-is for now and me having a crack at it in a followup?